collect_app/src/main/java/org/odk/collect/android/instancemanagement/InstanceSubmitter.kt

```diff
@@ -0,0 +1,130 @@
+package org.odk.collect.android.instancemanagement
+
+import org.odk.collect.analytics.Analytics
+import org.odk.collect.android.analytics.AnalyticsEvents
+import org.odk.collect.android.application.Collect
+import org.odk.collect.android.gdrive.GoogleAccountsManager
+import org.odk.collect.android.gdrive.GoogleApiProvider
+import org.odk.collect.android.gdrive.InstanceGoogleSheetsUploader
+import org.odk.collect.android.logic.PropertyManager
+import org.odk.collect.android.upload.FormUploadException
+import org.odk.collect.android.upload.InstanceServerUploader
+import org.odk.collect.android.upload.InstanceUploader
+import org.odk.collect.android.utilities.FormsRepositoryProvider
+import org.odk.collect.android.utilities.InstanceUploaderUtils
+import org.odk.collect.android.utilities.InstancesRepositoryProvider
+import org.odk.collect.android.utilities.WebCredentialsUtils
+import org.odk.collect.forms.FormsRepository
+import org.odk.collect.forms.instances.Instance
+import org.odk.collect.permissions.PermissionsProvider
+import org.odk.collect.settings.keys.ProjectKeys
+import org.odk.collect.shared.settings.Settings
+import timber.log.Timber
+
+class InstanceSubmitter(
+    private val analytics: Analytics,
+    private val formsRepository: FormsRepository,
+    private val googleAccountsManager: GoogleAccountsManager,
+    private val googleApiProvider: GoogleApiProvider,
+    private val permissionsProvider: PermissionsProvider,
+    private val generalSettings: Settings
+) {
+
+    @Throws(SubmitException::class)
+    fun submitInstances(toUpload: List<Instance>): HashMap<Instance, FormUploadException?> {
```

Comment:
This can just return `Map` instead of `HashMap`

Reply:
Fixed.